### PR TITLE
Migrate old page

### PR DESCRIPTION
**Pull request by mwilli20**
I feel like while this isn't done yet, it would be better to post up now to show the progress and the session planned for tomorrow will give some nice time to review the progress made so far.

### CODE REVIEW

File: coa_flask_app/__init__.py [hunk 1]
```diff
@@ -156,4 +156,53 @@ def locations_hierarchy() -> JSON:
...
         A json list of the locations hierarchy.
     """
-    return jsonify(locationsHierarchy=coa_logic.locations_hierarchy())
+    return jsonify(locationsHierarchy=site.locations_hierarchy())
+
+
+@APP.route('/getTLs')
+def get_tls() -> JSON:
+    """
+    The get tls route returns the all the TLs for the input drop down.
```
Comment (maithuvenkatesh) [inline coa_flask_app/__init__.py]: Minor:  Might be nice to change it to "The get tls route returns the all the team leads for the input drop down."

File: coa_flask_app/__init__.py [hunk 2]
```diff
@@ -156,4 +156,53 @@ def locations_hierarchy() -> JSON:
...
+
+    Returns:
+        A json list of the TLs.
```
Comment (maithuvenkatesh) [inline coa_flask_app/__init__.py]: Minor : "A json list of the team leads"

File: coa_flask_app/contribution.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    with Accessor() as db_handle:
+        db_handle.execute(query)
+        items = [(row[1],
+                  f'{row[2]}, {row[3]}[{row[0]}]')
+                 for row in db_handle.fetchall()]
+
+    trash_items: Dict[str, List[str]] = {}
+    for material, item in items:
+        if material in trash_items:
+            bisect.insort(trash_items[material], item)
```
Comment (maithuvenkatesh) [inline coa_flask_app/contribution.py]: Minor: Might want to add a comment that this returns the items sorted in alphabetical order.

File: coa_flask_app/site.py [hunk 4]
```diff
@@ -208,3 +218,75 @@ def locations_hierarchy() -> Dict[str, Dict[str, List[str]]]:
...
+                   location_name: str,
+                   start_date: str,
+                   end_date: str) -> List[Tuple[int, str, str, str, int]]:
+    """
+    Returns a list of tuples comprising of the item id, item name, category,
+    material, and quantity.
+
+    For example:
+    [
+        ...
```
Comment (maithuvenkatesh) [inline coa_flask_app/site.py]: Minor: Might be nice to complete the example.

File: coa_flask_app/site.py [hunk 5]
```diff
@@ -208,3 +218,75 @@ def locations_hierarchy() -> Dict[str, Dict[str, List[str]]]:
...
+        location_name: The name of the location.
+        start_date: The start date.
+        end_date: The end date.
+
+    Returns:
+        A list of item id, item name, category, material, quantity.
+    """
+    if location_category not in {'site_name', 'town', 'county'}:
+        return []
+
```
Comment (maithuvenkatesh) [inline coa_flask_app/site.py]: Is it worth validating that the end date is greater than the start date?

**Comment by mwilli20** [comment]
@maithuvenkatesh Finished your round of comments. @jerrylin1218 If you want to take a look. I would love now that CD isn't blocked, to be able to just make a ton of small PRs and post issues on stuff, rather then having these large PRs open with too much.

**Comment by mwilli20** [comment]
Not all of the code is pretty. I haven't been able to clean up everything I ported over. But for what is there or is being used, it 100% works. I haven't been able to test the parts not fully finished considering we don't have a front end for it. Just FYI.


The ugo parts:

https://github.com/CleanOceanAction/coa_flask_app/pull/29/files#diff-4de0ebd94fe510683d6be15f13c9ba9dR193
https://github.com/CleanOceanAction/coa_flask_app/pull/29/files#diff-4de0ebd94fe510683d6be15f13c9ba9dR203
https://github.com/CleanOceanAction/coa_flask_app/pull/29/files#diff-2fdce1f71cb758243a314b33433fefc2R67

**Comment by mwilli20** [comment]
Fixed merge conflict

**Pull request merged**